GameServerLib/ObjectManager.cs

```diff
@@ -134,16 +139,18 @@ public void Update(float diff)
                 var ai = u as IObjAiBase;
                 if (ai != null)
                 {
-                    var tempBuffs = ai.GetBuffs();
-                    for (int i = tempBuffs.Count - 1; i >= 0; i--)
+                    var tempBuffs = ai.Buffs.Get();  
```

Comment:
I don't know if `SimplePriorityQueue.ToList()` preserves order of elements, but I can't seem to implement a propper dequeue-loop since `BuffManager._buffsQueue` gets modified by changing `tempBuffs` (even when shadow-copy the `_buffsQueue`).

Reply:
It seems from here https://github.com/BlueRaja/High-Speed-Priority-Queue-for-C-Sharp/blob/master/Priority%20Queue/GenericPriorityQueue.cs that the order is like a perfect tree implemented by an array. That means that the order is completely random more or less, not even sorted by priority.

Reply:
Well, im stupid. Instead of using a prio queue internaly, i can just create it on demand...